homeassistant/components/recorder/migration.py

```diff
@@ -544,7 +544,7 @@ def _apply_update(instance, new_version, old_version):  # noqa: C901
                             # https://github.com/home-assistant/core/issues/56104
                             text(
                                 f"ALTER TABLE {table} CONVERT TO "
-                                "CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci LOCK=EXCLUSIVE"
+                                "CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci, LOCK=EXCLUSIVE"
```

Comment:
The alter table syntax doesn't document this very well
https://dev.mysql.com/doc/refman/5.6/en/alter-table.html?spm=a2c4g.11186623.2.23.cb0a2867KJiYg9

But there is an example here using a comma.
https://dev.mysql.com/doc/refman/5.6/en/innodb-online-ddl-operations.html

Reply:
```
> alter table states convert to CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci, LOCK=EXCLUSIVE;
Query OK, 0 rows affected (0.143 sec)
Records: 0  Duplicates: 0  Warnings: 0
```

Reply:
> The alter table syntax doesn't document this very well https://dev.mysql.com/doc/refman/5.6/en/alter-table.html?spm=a2c4g.11186623.2.23.cb0a2867KJiYg9
> 
> But there is an example here using a comma. https://dev.mysql.com/doc/refman/5.6/en/innodb-online-ddl-operations.html

Well, the grammar reads `alter_option [, alter_option]` which means that options should be comma-separated, and then `alter_option: {  CHARACTER SET [=] charset_name COLLATE [=] collation_name |  LOCK [=] {DEFAULT | NONE | SHARED | EXCLUSIVE} }` which means that `CHARACTER SET COLLATE` is the 1st option, and `LOCK` is the 2nd.